### PR TITLE
change format of nested error key and message

### PR DESCRIPTION
**Pull request by mrusakovich**
https://github.com/AaronLasseigne/active_interaction/issues/462

### CODE REVIEW

**Comment by h0jeZvgoxFepBQ2C** [comment]
Any idea @AaronLasseigne if this is mergeable?

File: spec/active_interaction/integration/array_interaction_spec.rb [hunk 1]
```diff
@@ -57,6 +57,24 @@ class CollectionProxy
...
+            integer :bar
+          end
+        end
+      end
+    end
+
+    let(:outcome) { interaction.run(foo: [{ bar: 'baz' }]) }
+
+    it 'sets error with nesting index' do
+      expect(outcome.errors).to include(:'foo.0.bar')
```
Comment (AaronLasseigne) [inline spec/active_interaction/integration/array_interaction_spec.rb]: It looks like Rails is doing this with brackets like `foo[0].bar`. https://blog.bigbinary.com/2016/07/07/errors-can-be-indexed-with-nested-attrbutes-in-rails-5.html

I try to stay in line with the Rails way as much as possible.

**Comment by AaronLasseigne** [comment]
This is a breaking change so it'll have to go into a major release. I was working on a v4 release but it's too much. I think I'm going to pair it back and we can add this to the new v4. When I have that ready, I'll rebase the merge onto that branch.

File: spec/active_interaction/integration/array_interaction_spec.rb (hunk 1 above)
Comment (mrusakovich) [reply spec/active_interaction/integration/array_interaction_spec.rb]: so, @AaronLasseigne , as far as I understand it should be changed from current form to `array_attribute[index].hash_key` form, am I right?

Comment (mrusakovich) [reply spec/active_interaction/integration/array_interaction_spec.rb]: error key format was changed according to the latest comment

**Comment by TylerRick** [comment]
Now that v4 is released, maybe we can look again into merging this?

**Comment by mrusakovich** [comment]
ye, it would be great to get any update on this, @AaronLasseigne, either close or merge)

**Comment by AaronLasseigne** [comment]
The upcoming 5.0.0 release reworks how errors like this are handled.

**Pull request closed without merging**